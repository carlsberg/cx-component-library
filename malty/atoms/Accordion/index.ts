export { Accordion, AccordionItem } from './Accordion';
export { AccordionColor, AccordionSize } from './Accordion.types';
export type { AccordionItemProps, AccordionProps } from './Accordion.types';
