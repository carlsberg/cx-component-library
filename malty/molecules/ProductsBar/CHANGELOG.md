# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.2.0 (2024-05-31)

### Bug Fixes

- Remove `eslint` warnings and errors ([#322](https://github.com/CarlsbergGBS/cx-component-library/issues/322)) ([4c7bc73](https://github.com/CarlsbergGBS/cx-component-library/commit/4c7bc737de8abb060919ac5ba16220a6d5d0ace9))
- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))
- **side nav:** [DSM-986] open nav when clicking on avatar ([#881](https://github.com/CarlsbergGBS/cx-component-library/issues/881)) ([8c5fc4b](https://github.com/CarlsbergGBS/cx-component-library/commit/8c5fc4b0f6a208d21ad5e9e0e04be946aa33e556))
- Update components to have a function declaration and remove unneeded fragments ([#502](https://github.com/CarlsbergGBS/cx-component-library/issues/502)) ([b470095](https://github.com/CarlsbergGBS/cx-component-library/commit/b4700954f4e1ae9c7e547f4de37ac2b15a9d55ea))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **DSM-248:** Headline HTML tags ([#184](https://github.com/CarlsbergGBS/cx-component-library/issues/184)) ([305f5e0](https://github.com/CarlsbergGBS/cx-component-library/commit/305f5e0fa92d5813f4877b338ad827a8c347d452))
- **DSM-362:** snapshots removed & unit test updates ([#307](https://github.com/CarlsbergGBS/cx-component-library/issues/307)) ([78423eb](https://github.com/CarlsbergGBS/cx-component-library/commit/78423eb5cfe04fe8a3a7af1f5af074ac6d35e235))
- **general:** [DSM-953] refactor icons implementation ([#892](https://github.com/CarlsbergGBS/cx-component-library/issues/892)) ([08d259f](https://github.com/CarlsbergGBS/cx-component-library/commit/08d259f476ce97e9db8ac41d5eb4773eabca0d37))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** Remove `react/jsx-props-no-spreading` comments ([#558](https://github.com/CarlsbergGBS/cx-component-library/issues/558)) ([823444c](https://github.com/CarlsbergGBS/cx-component-library/commit/823444ce0800bc7f49b584de59d247eb8497f559))
- Several component updates ([#185](https://github.com/CarlsbergGBS/cx-component-library/issues/185)) ([633c9f0](https://github.com/CarlsbergGBS/cx-component-library/commit/633c9f0a4432e9e3d08ff05156453660929c19a2))
- **sidenav:** [DSM-971] sidenav responsive behaviour ([#860](https://github.com/CarlsbergGBS/cx-component-library/issues/860)) ([0a84145](https://github.com/CarlsbergGBS/cx-component-library/commit/0a841458886a4314098890a787614414c4b12826))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
