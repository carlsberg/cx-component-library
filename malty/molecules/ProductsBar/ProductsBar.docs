---
labels: ['react', 'typescript', 'productsBar']
description: 'A Products bar component'
---