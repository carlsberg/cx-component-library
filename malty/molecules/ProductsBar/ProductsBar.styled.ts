import styled from 'styled-components';

export const StyledProductsBar = styled.div`
  width: ${({ theme }) => theme.sizes[`5xl`].value};
  height: 100%;
  background-color: ${({ theme }) => theme.colors.colours.default['digital-black'].value};
  border-width: ${({ theme }) => theme.borders['border-1px--solid']['border-width'].value};
  border-color: rgb(49, 69, 80);
  border-style: solid;
  box-sizing: border-box;
  padding: ${({ theme }) => `${theme.sizes.m.value} ${theme.sizes.l.value}`};
  overflow: hidden;
`;
export const StyledSystemWrapper = styled.div`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  max-width: calc(220px + 80px);
`;

export const StyledSystemMenu = styled.ul`
  width: 100%;
  position: relative;
  margin-bottom: ${({ theme }) => theme.sizes[`5xl`].value};
  list-style: none;
  padding: 0;
`;

export const StyledSystemOption = styled.li`
  width: ${({ theme }) => theme.sizes[`5xl`].value};
  height: ${({ theme }) => theme.sizes[`3xl`].value};
  list-style: none;
  cursor: pointer;
  position: relative;
  z-index: 100;
  &:not(:last-of-type) {
    margin-bottom: ${({ theme }) => theme.sizes[`2xs`].value};
  }
`;

export const StyledProfileBtn = styled(StyledSystemOption)`
  position: absolute;
  transform: translateY(-86px);
`;

export const StyledOptionIcon = styled.span`
  width: ${({ theme }) => theme.sizes[`3xl`].value};
  height: ${({ theme }) => theme.sizes[`3xl`].value};
  display: inline-block;
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  & svg {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
`;

export const StyledAvatar = styled.span`
  width: ${({ theme }) => theme.sizes.xl.value};
  height: ${({ theme }) => theme.sizes.xl.value};
  display: inline-block;
  position: absolute;
  margin-top: ${({ theme }) => theme.sizes.s.value};
  margin-bottom: ${({ theme }) => theme.sizes.m.value};
  left: 50%;
  transform: translateX(-50%);
`;

export const StyledProfileMenu = styled.div<{
  open: boolean;
}>`
  width: 100%;
  background-color: ${({ open }) => (open ? 'rgb(49,69,80)' : 'transparent')};
  position: absolute;
  transform: ${({ open }) => (open ? 'translateY(-100%)' : 'translateY(-84px)')};
  overflow: hidden;
  transition: all 0.2s ease-out;
  z-index: 200;
`;
export const StyledProfileHeader = styled.div`
  padding-left: ${({ theme }) => theme.sizes[`5xl`].value};
  top: 0;
  position: absolute;
  padding-top: ${({ theme }) => theme.sizes.m.value};
`;

export const StyledRoleLabel = styled.p`
  font-size: ${({ theme }) => theme.typography.desktop.text.small_default['font-size'].value};
  color: ${({ theme }) => theme.colors.colours.support[60].value};
  font-family: ${({ theme }) => theme.typography.desktop.text.small_default['font-family'].value};
  margin-top: ${({ theme }) => theme.sizes['4xs'].value};
`;

export const StyledProfileActions = styled.div<{
  open: boolean;
}>`
  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};
  & ul {
    padding-left: 0;
    margin-left: 0;
  }
`;

export const StyledProfileItem = styled.li`
  list-style: none;
  width: 100%;
  height: ${({ theme }) => theme.sizes[`3xl`].value};
  display: flex;
  padding-left: ${({ theme }) => theme.sizes[`4xl`].value};
  position: relative;
  align-items: center;
  & svg {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    left: ${({ theme }) => theme.sizes.l.value};
  }
  & span p {
    margin: 0;
  }
  & a {
    text-decoration: none;
  }
`;

export const StyledOverlay = styled.div<{
  open: boolean;
}>`
  width: 100%;
  height: 100%;
  z-index: 10;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  display: ${({ open }) => (open ? 'block' : 'none')};
  max-width: calc(220px + 80px);
  background-color: ${({ theme }) => theme.colors.colours.overlay['digital-black'][75].value};
`;
