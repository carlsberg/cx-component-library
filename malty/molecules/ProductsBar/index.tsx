export { ProductsBar } from './ProductsBar';
export { ProfileMenuType, SystemItemProps } from './ProductsBar.types';
