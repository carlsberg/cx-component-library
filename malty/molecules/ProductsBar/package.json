{
  "name": "@carlsberg/malty.molecules.products-bar",
  "version": "0.1.9",
  "main": "./lib/cjs/index.js",
  "module": "./lib/esm/index.js",
  "types": "./lib/esm/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/carlsberg/malty.git"
  },
  "publishConfig": {
    "@carlsberg:registry": "https://npm.pkg.github.com",
    "access": "public"
  },
  "scripts": {
    "build": "yarn build:esm && yarn build:cjs",
    "build:esm": "tsc",
    "build:cjs": "tsc --module commonjs --outDir lib/cjs",
    "test": "jest"
  },
  "dependencies": {
    "@carlsberg/malty.atoms.base-icon": "^0.0.5",
    "@carlsberg/malty.atoms.text": "^0.0.72",
    "@carlsberg/malty.molecules.avatar": "^0.1.6",
    "@carlsberg/malty.molecules.nav-list": "^0.1.9",
    "@carlsberg/malty.theme.malty-theme-provider": "^0.0.46",
    "core-js": "^3.0.0",
    "styled-components": "^5.3.6"
  },
  "devDependencies": {
    "@babel/runtime": "^7.12.18",
    "@carlsberg/malty.icons.data-transfer": "^0.1.3",
    "@carlsberg/malty.utils.test": "^0.0.21",
    "@testing-library/react": "^12.1.5",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^26.0.0",
    "@types/node": "12.20.4",
    "@types/react": "^17.0.8",
    "@types/react-dom": "^17.0.5",
    "@types/styled-components": "^5.1.26",
    "@types/testing-library__jest-dom": "5.9.5",
    "react-router-dom": "^6.2.2"
  },
  "peerDependencies": {
    "@carlsberg/malty.icons.carlsberg-filled": "*",
    "@carlsberg/malty.icons.menu": "*",
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0"
  }
}
