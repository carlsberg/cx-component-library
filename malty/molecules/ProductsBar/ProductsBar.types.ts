import { IconName } from '@carlsberggroup/malty.atoms.icon';

export type SystemItemProps = {
  icon: IconName;
  key?: string;
  href?: string;
  component?: React.ReactNode | JSX.Element;
};

export interface ProfileMenuProps {
  username: string;
  open: boolean;
  setProfileMenuOpen: (open: boolean) => void;
  children: React.ReactNode;
  userRole?: string;
}

type ProfileItemProps = {
  icon: IconName;
  key?: string;
  name?: string;
  href?: string;
  component?: React.ReactNode | JSX.Element;
};

export type ProfileMenuType = {
  username: string;
  userRole?: string;
  profileActions: ProfileItemProps[];
};

export interface ProductsBarProps {
  systemOptions: SystemItemProps[];
  profileMenu: ProfileMenuType;
  resetNavState: () => void;
  onToggleNav: () => void;
}
