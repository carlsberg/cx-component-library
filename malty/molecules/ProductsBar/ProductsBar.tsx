import { CloneIcon, IconColor, IconSize } from '@carlsberggroup/malty.atoms.base-icon';
import { Text, TextColor, TextStyle } from '@carlsberggroup/malty.atoms.text';
import { CarlsbergFilled } from '@carlsberggroup/malty.icons.carlsberg-filled';
import { Menu } from '@carlsberggroup/malty.icons.menu';
import { Avatar, AvatarSize } from '@carlsberggroup/malty.molecules.avatar';
import { LinkComponentProps } from '@carlsberggroup/malty.molecules.nav-list';
import { globalTheme as defaultTheme } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { RefObject, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';
import {
  StyledAvatar,
  StyledCollapseBtn,
  StyledOptionIcon,
  StyledOverlay,
  StyledProductsBar,
  StyledProfileActions,
  StyledProfileBtn,
  StyledProfileHeader,
  StyledProfileItem,
  StyledProfileMenu,
  StyledRoleLabel,
  StyledSystemMenu,
  StyledSystemOption,
  StyledSystemWrapper
} from './ProductsBar.styled';
import { ProductsBarProps, ProfileMenuProps } from './ProductsBar.types';

const LinkComponent = ({ component, href, children, componentProps }: LinkComponentProps) => {
  const CustomComponent = (component as keyof JSX.IntrinsicElements) || null;

  return component ? <CustomComponent {...componentProps}>{children}</CustomComponent> : <a href={href}>{children}</a>;
};

const useClickOutside = (
  ref: RefObject<HTMLDivElement>,
  open: boolean,
  setProfileMenuOpen: (open: boolean) => void
) => {
  /**
   * Detect click outside of element event
   */

  useEffect(() => {
    function handleClickOutside(event: MouseEvent) {
      const clickedElem = event.target as Element;
      if (ref.current && !ref.current.contains(clickedElem)) {
        if (!open) {
          return;
        }
        setProfileMenuOpen(false);
      }
    }

    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [open]);
};

const ProfileMenu = ({
  open,
  setProfileMenuOpen,
  username,
  userRole,
  children,
  onOpenNav,
  isNavOpen = false
}: ProfileMenuProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  const profileMenuRef = React.useRef<HTMLDivElement>(null);

  useClickOutside(profileMenuRef, open, setProfileMenuOpen);

  const toggleProfileMenu = () => {
    onOpenNav();
    setProfileMenuOpen(!open);
  };

  return (
    <StyledProfileMenu open={open} ref={profileMenuRef} theme={theme}>
      <StyledSystemOption theme={theme}>
        <StyledOptionIcon theme={theme}>
          <StyledAvatar theme={theme} data-testid="avatar">
            <Avatar userName={username} size={AvatarSize.Medium} onClick={toggleProfileMenu} />
          </StyledAvatar>
        </StyledOptionIcon>
      </StyledSystemOption>
      <StyledProfileActions open={open}>
        <StyledProfileHeader theme={theme}>
          <Text textStyle={TextStyle.MediumDefault} color={TextColor.White}>
            {username}
          </Text>
          <StyledRoleLabel theme={theme}>{userRole}</StyledRoleLabel>
        </StyledProfileHeader>
        {children}
      </StyledProfileActions>
    </StyledProfileMenu>
  );
};

export const ProductsBar = ({
  systemOptions,
  profileMenu,
  resetNavState,
  onToggleNav,
  onOpenNav,
  isNavOpen
}: ProductsBarProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  const [profileMenuOpen, setProfileMenuOpen] = useState(false);
  const { username, userRole, profileActions } = profileMenu;
  const {
    component: singleItemComponent,
    href: singleItemHref,
    icon: singleItemIcon,
    ...singleItemCustomProps
  } = profileActions[0];

  const singleItemCompProps = { ...singleItemCustomProps };

  return (
    <StyledProductsBar theme={theme}>
      <StyledOverlay open={profileMenuOpen} theme={theme} />
      <CarlsbergFilled color={IconColor.White} size={IconSize.Medium} />
      <StyledCollapseBtn onClick={onToggleNav} data-testid="collapse-button">
        <Menu color={IconColor.White} size={IconSize.Medium} />
      </StyledCollapseBtn>
      <StyledSystemWrapper theme={theme}>
        <StyledSystemMenu theme={theme} data-testid="system-options">
          {systemOptions?.map((option, index) => {
            const { icon, href, component, ...customProps } = option;
            const componentProps = { ...customProps };

            return (
              <StyledSystemOption theme={theme} onClick={resetNavState} key={option.key || `systemOption${index}`}>
                <StyledOptionIcon theme={theme}>
                  <LinkComponent component={component} href={href} componentProps={componentProps}>
                    <CloneIcon icon={icon} color={IconColor.White} size={IconSize.Medium} />
                  </LinkComponent>
                </StyledOptionIcon>
              </StyledSystemOption>
            );
          })}
        </StyledSystemMenu>

        {profileActions.length === 1 && (
          <StyledProfileBtn theme={theme}>
            <StyledOptionIcon theme={theme}>
              <LinkComponent component={singleItemComponent} href={singleItemHref} componentProps={singleItemCompProps}>
                <StyledAvatar theme={theme} data-testid="avatar">
                  <Avatar userName={username} size={AvatarSize.Medium} onClick={resetNavState} />
                </StyledAvatar>
              </LinkComponent>
            </StyledOptionIcon>
          </StyledProfileBtn>
        )}

        {profileActions.length > 1 && (
          <ProfileMenu
            open={profileMenuOpen}
            username={username}
            userRole={userRole}
            setProfileMenuOpen={setProfileMenuOpen}
            onOpenNav={onOpenNav}
            isNavOpen={isNavOpen}
          >
            <ul data-testid="profile-options">
              {profileActions.map((action, index) => {
                const { name = 'item', icon, component, href, ...customProps } = action;
                const componentProps = { ...customProps };

                return (
                  <StyledProfileItem theme={theme} onClick={resetNavState} key={action.key || `profileOption${index}`}>
                    <LinkComponent component={component} href={href} componentProps={componentProps}>
                      <CloneIcon icon={icon} color={IconColor.White} size={IconSize.Medium} />
                      <Text textStyle={TextStyle.MediumSmallDefault} color={TextColor.White}>
                        {name}
                      </Text>
                    </LinkComponent>
                  </StyledProfileItem>
                );
              })}
            </ul>
          </ProfileMenu>
        )}
      </StyledSystemWrapper>
    </StyledProductsBar>
  );
};
