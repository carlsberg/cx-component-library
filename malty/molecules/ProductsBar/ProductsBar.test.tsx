import { IconName } from '@carlsberggroup/malty.atoms.icon';
import { render } from '@carlsberggroup/malty.utils.test';
import { fireEvent, screen, within } from '@testing-library/react';
import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import { ProductsBar } from './ProductsBar';

const systemOptionsMock = [
  { icon: IconName.DataTransfer, href: '/iframe.html' },
  { icon: IconName.DataTransfer, component: Link, to: '/item2' }
];

const profileMenuMock = {
  username: 'Maria Snow',
  userRole: 'Market director',
  profileActions: [
    { name: 'User profile', icon: IconName.DataTransfer, component: Link, to: '/profile' },
    { name: 'Sign out', icon: IconName.DataTransfer, component: Link, to: '/sign-out' }
  ]
};

const singleOptionConfig = {
  username: 'Maria Snow',
  userRole: 'Market director',
  profileActions: [{ name: 'User profile', icon: IconName.DataTransfer, component: Link, to: '/profile' }]
};

const resetNavState = jest.fn();

describe('Products bar component', () => {
  it('renders with correct number of system options', () => {
    render(
      <BrowserRouter>
        <ProductsBar systemOptions={systemOptionsMock} profileMenu={profileMenuMock} resetNavState={resetNavState} />
      </BrowserRouter>
    );
    const options = screen.getByTestId('system-options');
    const { getAllByRole } = within(options);
    const items = getAllByRole('listitem');
    expect(items).toHaveLength(2);
  });

  it('opens profile menu when there is more than on action configured', () => {
    render(
      <BrowserRouter>
        <ProductsBar systemOptions={systemOptionsMock} profileMenu={profileMenuMock} resetNavState={resetNavState} />
      </BrowserRouter>
    );
    const avatarContainer = screen.getByTestId('avatar');

    fireEvent.click(avatarContainer);
    const profileMenu = screen.getByTestId('profile-options');
    expect(profileMenu).toBeDefined();
  });

  it("doesn't render profile menu when there is only one action configured", () => {
    render(
      <BrowserRouter>
        <ProductsBar systemOptions={systemOptionsMock} profileMenu={singleOptionConfig} resetNavState={resetNavState} />
      </BrowserRouter>
    );
    const avatarContainer = screen.getByTestId('avatar');

    fireEvent.click(avatarContainer);
    const profileMenu = screen.queryByTestId('profile-options');
    expect(profileMenu).toBeNull();
  });
});
