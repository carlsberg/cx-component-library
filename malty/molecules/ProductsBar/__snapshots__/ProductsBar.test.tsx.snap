// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products bar component matches snapshot 1`] = `
<div
  className="sc-hKMtZM ubCAJ"
>
  <div
    className="sc-breuTD cosMe"
    open={false}
  />
  <svg
    className="sc-gsnTZi iOMLMs"
    color="white"
    data-testid="svg-component"
    size="24px"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M12 6.753a1.33 1.33 0 01-1.341-1.32V2.757C10.659 1.79 9.867 1 8.877 1c-.41 0-.784.153-1.087.388a4.256 4.256 0 01-4.149.61C2.433 1.533 1.023 1.981.353 3.124a2.618 2.618 0 00.65 3.372 4.12 4.12 0 011.542 3.842c-.055.38-.002.774.204 1.13a1.801 1.801 0 002.433.64l2.355-1.34a1.348 1.348 0 011.826.486 1.31 1.31 0 01-.495 1.8l-2.35 1.339a1.737 1.737 0 00-.646 2.398c.204.355.53.594.883.734 1.375.546 2.387 1.76 2.608 3.24A2.653 2.653 0 0012 23c1.34 0 2.438-.967 2.629-2.234.229-1.48 1.243-2.695 2.611-3.241.36-.14.682-.38.887-.734a1.736 1.736 0 00-.654-2.398l-2.351-1.34a1.305 1.305 0 01-.486-1.8c.369-.63 1.187-.85 1.824-.486l2.35 1.342a1.809 1.809 0 002.44-.642c.2-.355.256-.75.2-1.129a4.117 4.117 0 011.538-3.842c1.014-.8 1.327-2.226.654-3.372-.672-1.143-2.073-1.591-3.288-1.126a4.252 4.252 0 01-4.148-.61 1.765 1.765 0 00-2.87 1.37v2.674c0 .728-.6 1.32-1.336 1.32"
      fillRule="evenodd"
    />
  </svg>
  <div
    className="sc-eCYdqJ bZezBM"
  >
    <ul
      className="sc-jSMfEi dJEoXJ"
      data-testid="system-options"
    >
      <li
        className="sc-gKXOVf kuyIwz"
        onClick={[Function]}
      >
        <span
          className="sc-ftvSup dtcHEr"
        >
          <a
            href="/iframe.html"
          >
            <svg
              className="sc-gsnTZi iOMLMs"
              color="white"
              data-testid="svg-component"
              size="24px"
              version="1.1"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <path
                d="M16 2a1 1 0 01.117 1.993L16 4H7a1 1 0 00-.993.883L6 5v14a1 1 0 00.883.993L7 20h9a1 1 0 01.117 1.993L16 22H7a3 3 0 01-2.995-2.824L4 19V5a3 3 0 012.824-2.995L7 2zm-4 13a1 1 0 01.117 1.993L12 17H9a1 1 0 01-.117-1.993L9 15zm5.293-6.667a1 1 0 011.414 0l3 3a1 1 0 010 1.414l-3 3a1 1 0 01-1.414-1.414l2.293-2.293-2.293-2.293a1 1 0 010-1.414zM12 11a1 1 0 01.117 1.993L12 13H9a1 1 0 01-.117-1.993L9 11zm0-4a1 1 0 01.117 1.993L12 9H9a1 1 0 01-.117-1.993L9 7z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </span>
      </li>
      <li
        className="sc-gKXOVf kuyIwz"
        onClick={[Function]}
      >
        <span
          className="sc-ftvSup dtcHEr"
        >
          <a
            href="/item2"
            onClick={[Function]}
          >
            <svg
              className="sc-gsnTZi iOMLMs"
              color="white"
              data-testid="svg-component"
              size="24px"
              version="1.1"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <path
                d="M16 2a1 1 0 01.117 1.993L16 4H7a1 1 0 00-.993.883L6 5v14a1 1 0 00.883.993L7 20h9a1 1 0 01.117 1.993L16 22H7a3 3 0 01-2.995-2.824L4 19V5a3 3 0 012.824-2.995L7 2zm-4 13a1 1 0 01.117 1.993L12 17H9a1 1 0 01-.117-1.993L9 15zm5.293-6.667a1 1 0 011.414 0l3 3a1 1 0 010 1.414l-3 3a1 1 0 01-1.414-1.414l2.293-2.293-2.293-2.293a1 1 0 010-1.414zM12 11a1 1 0 01.117 1.993L12 13H9a1 1 0 01-.117-1.993L9 11zm0-4a1 1 0 01.117 1.993L12 9H9a1 1 0 01-.117-1.993L9 7z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </span>
      </li>
    </ul>
    <div
      className="sc-jqUVSM cVlWPk"
      open={false}
    >
      <li
        className="sc-gKXOVf kuyIwz"
        onClick={[Function]}
      >
        <span
          className="sc-ftvSup dtcHEr"
        >
          <span
            className="sc-papXJ iHFpNU"
            data-testid="avatar"
          />
        </span>
      </li>
      <div
        className="sc-crXcEl eCrVYO"
        open={false}
      >
        <div
          className="sc-kDDrLX kGIrKz"
        >
          <span
            className="sc-bczRLJ qaxFj"
          >
            <p
              className="sc-dkzDqf eSqqkM"
              color="white"
            >
              Maria Snow
            </p>
          </span>
          <p
            className="sc-iqcoie ctpUi"
          >
            Market director
          </p>
        </div>
        <ul
          data-testid="profile-options"
        >
          <li
            className="sc-evZas bzEkrF"
            onClick={[Function]}
          >
            <a
              href="/profile"
              onClick={[Function]}
            >
              <svg
                className="sc-gsnTZi iOMLMs"
                color="white"
                data-testid="svg-component"
                size="24px"
                version="1.1"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <path
                  d="M16 2a1 1 0 01.117 1.993L16 4H7a1 1 0 00-.993.883L6 5v14a1 1 0 00.883.993L7 20h9a1 1 0 01.117 1.993L16 22H7a3 3 0 01-2.995-2.824L4 19V5a3 3 0 012.824-2.995L7 2zm-4 13a1 1 0 01.117 1.993L12 17H9a1 1 0 01-.117-1.993L9 15zm5.293-6.667a1 1 0 011.414 0l3 3a1 1 0 010 1.414l-3 3a1 1 0 01-1.414-1.414l2.293-2.293-2.293-2.293a1 1 0 010-1.414zM12 11a1 1 0 01.117 1.993L12 13H9a1 1 0 01-.117-1.993L9 11zm0-4a1 1 0 01.117 1.993L12 9H9a1 1 0 01-.117-1.993L9 7z"
                  fillRule="evenodd"
                />
              </svg>
              <span
                className="sc-bczRLJ qaxFj"
              >
                <p
                  className="sc-dkzDqf jdtZuS"
                  color="white"
                >
                  User profile
                </p>
              </span>
            </a>
          </li>
          <li
            className="sc-evZas bzEkrF"
            onClick={[Function]}
          >
            <a
              href="/sign-out"
              onClick={[Function]}
            >
              <svg
                className="sc-gsnTZi iOMLMs"
                color="white"
                data-testid="svg-component"
                size="24px"
                version="1.1"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <path
                  d="M16 2a1 1 0 01.117 1.993L16 4H7a1 1 0 00-.993.883L6 5v14a1 1 0 00.883.993L7 20h9a1 1 0 01.117 1.993L16 22H7a3 3 0 01-2.995-2.824L4 19V5a3 3 0 012.824-2.995L7 2zm-4 13a1 1 0 01.117 1.993L12 17H9a1 1 0 01-.117-1.993L9 15zm5.293-6.667a1 1 0 011.414 0l3 3a1 1 0 010 1.414l-3 3a1 1 0 01-1.414-1.414l2.293-2.293-2.293-2.293a1 1 0 010-1.414zM12 11a1 1 0 01.117 1.993L12 13H9a1 1 0 01-.117-1.993L9 11zm0-4a1 1 0 01.117 1.993L12 9H9a1 1 0 01-.117-1.993L9 7z"
                  fillRule="evenodd"
                />
              </svg>
              <span
                className="sc-bczRLJ qaxFj"
              >
                <p
                  className="sc-dkzDqf jdtZuS"
                  color="white"
                >
                  Sign out
                </p>
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
